datalad/distributed/create_sibling_ria.py

```diff
@@ -0,0 +1,586 @@
+"""Create a sibling in a RIA store"""
+
+__docformat__ = 'restructuredtext'
+
+
+import logging
+import subprocess
+
+from datalad.interface.common_opts import (
+    recursion_flag,
+    recursion_limit
+)
+from datalad.interface.base import (
+    Interface,
+    build_doc,
+)
+from datalad.interface.results import (
+    get_status_dict,
+)
+from datalad.interface.utils import eval_results
+from datalad.support.param import Parameter
+from datalad.support.constraints import (
+    EnsureNone,
+    EnsureStr,
+    EnsureBool,
+    EnsureChoice
+)
+from datalad.distribution.dataset import (
+    EnsureDataset,
+    datasetmethod,
+    require_dataset,
+)
+from datalad.utils import (
+    Path,
+    quote_cmdlinearg,
+    rmtree,
+)
+from datalad.support.exceptions import (
+    CommandError
+)
+from datalad.support.gitrepo import (
+    GitRepo
+)
+from datalad.core.distributed.clone import (
+    decode_source_spec
+)
+from datalad.log import log_progress
+from datalad.distributed.ria_utils import (
+    get_layout_locations,
+    verify_ria_url,
+)
+
+lgr = logging.getLogger('datalad.distributed.create_sibling_ria')
+
+
+@build_doc
+class CreateSiblingRia(Interface):
+    """Creates a sibling to a dataset in a RIA store
+
+    This creates a representation of a dataset in a ria-remote compliant
+    storage location. For access to it two siblings are configured for the
+    dataset by default. A "regular" one and a RIA remote (git-annex
+    special remote). Furthermore, the former is configured to have a
+    publication dependency on the latter. If not given a default name for
+    the RIA remote is derived from the sibling's name by appending "-ria".
+
+    The store's base path currently is expected to either:
+
+      - not yet exist or
+      - be empty or
+      - have a valid "ria-layout-version" file and an "error_logs" directory.
+
+    In the first two cases, said file and directory are created by this
+    command. Alternatively you can manually create the third case, of course.
+    Please note, that "ria-layout-version" needs to contain a line stating the
+    version (currently "1") and optionally enable error logging (append a pipe
+    symbol and an "l"  in that case). Currently, this line MUST end with a newline!
+
+    Error logging will create files in the "error_log" directory whenever the
+    RIA special remote (storage sibling) raises an exception, storing the
+    python traceback of it. The logfiles are named according to the scheme
+    <dataset id>.<annex uuid of the remote>.log showing 'who' ran into this
+    issue with what dataset. Since this logging can potentially leak personal
+    data (like local file paths for example) it can be disabled from the client
+    side via "annex.ria-remote.<RIAREMOTE>.ignore-remote-config".
+
+    Todo
+    ----
+    Where to put the description of a RIA store (see below)?
+
+    The targeted layout of such a store is a tree of datasets, starting at the
```

Comment:
Make it "tree of directories containing datasets at the lowest level", since otherwise suggests hierarchy of Datasets